### ISSUE TITLE
`tqdm.write()` overwrite closed nested progress bar

### ISSUE THREAD

**Issue by Hiestaa** [tqdm/tqdm#719]
I have looked around for a solution to this problem with no success. I have found issue https://github.com/tqdm/tqdm/issues/398 which seems to report a similar issue, but the problem there is related to open nested progress bars. My issue has to do with the `tqdm.write()` calls **after** the progress bar is closed. More details below:

## Versions 

* TQDM v4.31.1
* Python v3.6.8
* OS: MacOSX Mojave v10.14.3 (18D42)
* `sys.version`: 3.6.8 (default, Apr 17 2019, 10:09:55) [GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.46.3)]

## Description

I am running into a nested progress bar clobbering issue when using `tqdm.write()` past the closing of the progress bar. Here is a minimal code example to reproduce the issue:

```
import time
from tqdm import trange, tqdm

for x in trange(5):
    for y in trange(5):
        time.sleep(0.01)
        tqdm.write('x={}, y={}'.format(x, y))


tqdm.write('x' * 50)
tqdm.write('x' * 50)
```

Here is the output I'm seeing in the terminal after the process is complete
```
x=0, y=0
x=0, y=1
x=0, y=2
x=0, y=3
x=0, y=4
x=1, y=0
x=1, y=1
x=1, y=2
x=1, y=3
x=1, y=4
x=2, y=0
x=2, y=1
x=2, y=2
x=2, y=3
x=2, y=4
x=3, y=0
x=3, y=1
x=3, y=2
x=3, y=3
x=3, y=4
x=4, y=0
x=4, y=1
x=4, y=2
x=4, y=3
x=4, y=4
100%|█████████████████████████████████████████████████████████| 5/5 [00:00<00:00, 17.10it/s]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx████████████| 5/5 [00:00<00:00, 85.86it/s]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

* Expected behavior: The first line of `xxxxxxx....` should be printed *after* the closed nested progress bar.
* Actual behavior: The first line of `xxxxx....` clobbers the completed and closed progress bar, resulting in less than ideal and quite confusing log output. 

## Additional notes:
* if using 3 or more nested progress bars, only the *first* progress bar will be remain intact. All subsequent, closed progress bar will be clobbered. Example output from slightly modified code above to include a third nested loop:
```
x=4, y=3, z=4
x=0, y=4, z=0
x=1, y=4, z=1
x=2, y=4, z=2
x=3, y=4, z=3
x=4, y=4, z=4
100%|█████████████████████████████████████████████████████████| 5/5 [00:01<00:00,  3.43it/s]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx████████████| 5/5 [00:00<00:00, 17.21it/s]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx████████████| 5/5 [00:00<00:00, 86.46it/s]
```

* Using manual creation and closing of the progress bar (with `pbar = tqdm(range(5), total=5)` and `pbar.close()`) did not change the behaviour (not that it should be)

**Comment by casperdcl** [comment]
Yes, this is a know issue where you have to `write/print('\n' * number_of_nested_bars)` after completion before `write/print('x' * 50)`

**Comment by Hiestaa** [comment]
Is the known issue reported anywhere? If this is a duplicate I apologize for not having found it. Please link and close this issue if you have a link handy.

Thank you for the workaround suggestion, I'll go ahead and do this. However, this seems to leave blank lines in the logs which isn't ideal, even when only printing a single `\n` character:

```
import time
from tqdm import trange, tqdm

for x in trange(5):
    for y in trange(5):
        tqdm.write('x={}, y={}, z={}'.format(x, y, x))

tqdm.write('\n')
tqdm.write('x' * 50)
tqdm.write('x' * 50)
```

Output:
```
[...]
x=4, y=3, z=4
x=4, y=4, z=4
100%|████████████████████████████████████████████████████████████| 5/5 [00:00<00:00,  8.57it/s]
100%|████████████████████████████████████████████████████████████| 5/5 [00:00<00:00, 42.27it/s]

xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```
(the same happens with `print()`, btw)


I wonder how *one* `tqdm.write('\n')` can cause two blank lines to be added (the one that would clobbed the progress bar, and the blank line that separates the bar from the line of Xs). 

That still works fine enough for now, thanks.

**Comment by a-gn** [comment]
> Yes, this is a know issue where you have to `write/print('\n' * number_of_nested_bars)` after completion before `write/print('x' * 50)`

Is there a way to know how many progress bars are currently open? I can't predict this easily in my case.

**Comment by casperdcl** [comment]
you could do `print('\n' * len(tqdm._instances))`
